src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -357,15 +359,40 @@ void SetArrayItemNode(object d)
                         }
                         else if (node is ValueExecutionNode valueNode)
                         {
-                            node.Result = valueNode.GraphType.Serialize(d)
-                                ?? throw new InvalidOperationException($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}' for list index {index}.");
+                            bool defaultException = false;
+                            try
+                            {
+                                object result = valueNode.GraphType.Serialize(d);
+                                if (result == null)
+                                {
+                                    defaultException = true;
+                                    if (listType.ResolvedType is NonNullGraphType)
+                                        throw new InvalidOperationException($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}' for list index {index}.");
+                                    else
+                                        throw new ExecutionError($"Unable to serialize '{d}' to '{valueNode.GraphType.Name}'.");
```

Comment:
I do not understand this.  Why the difference?  The problem is that `Serialize` was unable to serialize a value returned from the field resolver.  For example, "abc" was returned for an `IntGraphType`.  It does not matter if it is a non-null list type or not.  The problem lies in the resolver; it did not return data in the proper format.

`ExecutionError` indicates there is a client-side problem.  That is not the case as described above; the returned value from the field resolver cannot be coerced to the proper type.  Therefor, it is an `InvalidOperationException` either way.  And there is no reason to have different error messages.

Reply:
`ExecutionError` because in that case it will be used only for `SetNodeError`. `SetNodeError` takes `ExecutionError`, not `Exception`. OK?